handlers.js

```diff
@@ -153,63 +153,116 @@ function maxAmountGuard(proceed) {
   }
 }
 
+// Ensure a request has minimum required params
+function hasParams(params, required) {
+  for (var i = 0; i < required.length; i++) {
+    if (typeof params[required[i]] !== 'string') {
+      return false
+    }
+  }
+  return true
+}
+
 // Peer API Implementation
 const handlers = {
   getSenderSideQuote: tradingPairGuard(
     maxAmountGuard(function(params, callback) {
-      callback(
-        null,
-        createQuote({
-          senderParam: priceSell(params),
-          ...params,
-        }),
-      )
+      const required = ['signerParam', 'signerToken', 'signerKind', 'senderToken', 'senderKind']
+      if (hasParams(params, required)) {
+        callback(
+          null,
+          createQuote({
+            senderParam: priceSell(params),
+            ...params,
+          }),
+        )
+      } else {
+        callback({
+          code: -33604,
+          message: `Require ${required.join(', ')}`,
+        })
+      }
     }),
   ),
   getSignerSideQuote: tradingPairGuard(
     maxAmountGuard(function(params, callback) {
-      callback(
-        null,
-        createQuote({
-          signerParam: priceBuy(params),
-          ...params,
-        }),
-      )
+      const required = ['senderParam', 'senderToken', 'senderKind', 'signerToken', 'signerKind']
+      if (hasParams(params, required)) {
+        callback(
+          null,
+          createQuote({
+            signerParam: priceBuy(params),
+            ...params,
+          }),
+        )
+      } else {
+        callback({
+          code: -33604,
+          message: `Require ${required.join(', ')}`,
+        })
+      }
     }),
   ),
   getMaxQuote: tradingPairGuard(function(params, callback) {
     const signerParam = getMaxParam({ signerParam: params.signerParam, signerToken: params.signerToken })
-    callback(
-      null,
-      createQuote({
-        signerParam: signerParam.toString(),
-        senderParam: priceSell({ signerParam, ...params }),
-        ...params,
-      }),
-    )
-  }),
-  getSenderSideOrder: tradingPairGuard(
-    maxAmountGuard(async function(params, callback) {
+    const required = ['signerToken', 'signerKind', 'senderToken', 'senderKind']
```

Comment:
So this adds it as a required field on the request, correct? 
Currently in the docs we don't have it on the request: https://docs.airswap.io/instant-beta/running-makers#getmaxquote
But we do have it on the response: https://docs.airswap.io/instant-beta/orders-and-signatures#quotes-versus-orders

Happy to make this change though.

Reply:
That's a misunderstanding, `kind` is not required in requests and is up to the signer to determine.

Reply:
Updated.